packages/platforms/legacy-adapter-player/src/android-legacy-api/api.ts

```diff
@@ -0,0 +1,80 @@
+import { Msg } from './msg';
+
+import { inbound } from './inbound';
+import { outbound } from './outbound';
+
+interface PlayerData {
+  player_version: string;
+  model: string;
+  platform: string;
+  uuid: string;
+  android_version: string;
+}
+
+const msg = new Msg(window.parent, inbound, outbound);
+
+const SIMPLE_ACTIONS_TIMEOUT_MS = 5000;
+// Return a promise that rejects after a timeout
+const timeout = (ms = SIMPLE_ACTIONS_TIMEOUT_MS) =>
+  new Promise((_, reject) =>
+    setTimeout(
+      () => reject(new Error('Request to Android player api timed out')),
+      ms
+    )
+  );
+
+export async function getPlayerData(): Promise<PlayerData> {
+  console.log('Android api: getPlayerData');
+  return Promise.race([
+    outbound.getPlayerData(),
+    timeout(),
+  ]) as unknown as PlayerData;
+}
+
+export async function setItem(key: string, value: string): Promise<void> {
+  console.log('Android api: setItem', key, value);
+  await Promise.race([outbound.set(key, value), timeout()]);
```

Comment:
Hmm. The timeout may actually cause more problems than it solves... Since this is the legacy adapter we're just firing the event and have no means to implement an abortcontroller. Perhaps I'll just remove the timeouts here...

Reply:
Removed timeouts